addons/product_expiry/tests/test_stock_lot.py

```diff
@@ -80,51 +80,54 @@ def test_00_stock_production_lot(self):
         self.env['stock.lot']._alert_date_exceeded()
 
         # check a new activity has been created
-        activity_id = self.env.ref('product_expiry.mail_activity_type_alert_date_reached').id
-        activity_count = self.env['mail.activity'].search_count([
-            ('activity_type_id', '=', activity_id),
-            ('res_model_id', '=', self.env.ref('stock.model_stock_lot').id),
-            ('res_id', '=', self.lot1_productAAA.id)
-        ])
-        self.assertEqual(activity_count, 1, 'No activity created while there should be one')
+        activity_type = self.env.ref('product_expiry.mail_activity_type_alert_date_reached', raise_if_not_found=False)
+        if activity_type:
+            activity_count = self.env['mail.activity'].search_count([
+                ('activity_type_id', '=', activity_type.id),
+                ('res_model_id', '=', self.env.ref('stock.model_stock_lot').id),
+                ('res_id', '=', self.lot1_productAAA.id)
+            ])
+            self.assertEqual(activity_count, 1, 'No activity created while there should be one')
 
         # run the scheduler a second time
         self.env['stock.lot']._alert_date_exceeded()
```

Comment:
I guess it still calls the method, even if it does absolutely nothing afterwards 😄 

Shouldn't we consider that, in a test environment, our master data activity types exist? Not sure of the added benefit.